v2/dtos/requests/event.go

```diff
@@ -19,9 +19,7 @@ import (
 // https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/AddEventRequest
 type AddEventRequest struct {
 	common.BaseRequest `json:",inline"`
-	DeviceName         string             `json:"deviceName" validate:"required"`
-	Origin             int64              `json:"origin" validate:"required"`
-	Readings           []dtos.BaseReading `json:"readings"`
+	Event              dtos.Event `json:"event"`
```

Comment:
Add ` validate:"required,dive"`

Reply:
Thanks. Updated to `validate:"required"`. No need for the `dive` tag here as the inner `Event` is not a slice, array or map.